### ISSUE TITLE
ncurses 5.9 fails to compile (gcc > 5.1)

### ISSUE THREAD

**Issue by yhoogstrate** [galaxyproject/tools-iuc#477]
On my local instance installation of package_r_edger_3_11_0 fails because compilation of package_ncurses_5_9 fails with the following error:

```
STDERR
In file included from ../ncurses/curses.priv.h:283:0,
                 from ../ncurses/lib_gen.c:19:
_4238.c:835:15: error: expected ‘)’ before ‘int’
../include/curses.h:1594:56: note: in definition of macro ‘mouse_trafo’
 #define mouse_trafo(y,x,to_screen) wmouse_trafo(stdscr,y,x,to_screen)
                                                        ^
make[1]: *** [../obj_s/lib_gen.o] Error 1
make: *** [all] Error 2
```

It seems to be a GCC >5.1 specific issue: https://github.com/cloudius-systems/osv/issues/634

This is the third issue caused by specific versions of GCC. (https://github.com/galaxyproject/tools-iuc/issues/466 and https://github.com/galaxyproject/tools-iuc/pull/306 / https://github.com/galaxyproject/tools-iuc/issues/304 )
I know we have a C/GCC compiler as requirement for installation, but maybe it's time to consider to have a C/C++ compiler, gcc or clang, as a tool dependency in iuc?

To solve the issue for edgeR, could I upgrade the R version from 3.1.2 to 3.2.1 (depending on ncursus 6.0 which is not affected) or may this break test cases?

**Comment by mvdbeek** [comment]
Well, in this case we do get a patch that likely resolves the issue.
I don't know what the implications would be of having a gcc package.

We could be running into hard to debug problems when mixing packages copiled by different gcc versions ...
Another thing would be people that need specific gcc versions that are fixed on their cluster.
https://stackoverflow.com/questions/23895081/can-you-mix-c-compiled-with-different-versions-of-the-same-compiler

It is certainly an interesting option.

**Comment by bgruening** [comment]
Not sure if I posted it somewhere already, but I'm on @yhoogstrate side here. This is a very big problem and it will become bigger next year, especially if we will need to support Clang as well. 
Imho, we should consider to offer a base easy_build toolchain or a Guix build which takes care of you dependencies. This toolchain can than be updated and maintained over a longer time period.

@yhoogstrate I would be fine with updating your dependencies to newer versions to get this fixed.

**Comment by mvdbeek** [comment]
Well, I'm just afraid of having to maintain a gcc build version for everything, assuming that simple mixing of packages won't work easily. But I guess we should get our hands dirty and try to see how it goes. Maybe it turns out to save us time.

**Comment by yhoogstrate** [comment]
> Well, I'm just afraid of having to maintain a gcc build version for everything

If we would go down that way we would definitely be doing it wrong! 
The advantage of having a compiler dependency shall not be that we won't encounter compiler specific issues anymore, but that we encounter them in the development/testing phase rather than in the production phase. If a package doesn't compile with the compiler present in iuc-tools, we know this upfront and the tool won't get into the toolshed until it's solved (preferably upstream). If something like that happens there must be a very good reason to add another compiler since C and C++ have standards.

**Comment by bgruening** [comment]
Fixed in #486
